server/controllers/index.ts

```diff
@@ -70,6 +70,50 @@ export abstract class BaseController<
     req.sessionModel.set('csrf-secret', res.locals.csrfToken)
     next()
   }
+
+  /**
+   * Builds an object of values for all fields and steps from session or, alternatively,
```

Comment:
Isn't this what Form Wizard does anyway? 🤔 

I can see values from previous steps but I'm probably missing what this is about.

Reply:
i found that `req.form.values` doesn’t just contain what you expect, but other things too (like CSRF token) plus it’s not necessarily valid data. i wonder if this is overkill, but retrieving things from the session model felt safer to me.

once we do more manual testing, we can maybe drop this in favour of values…?

Reply:
yeah, I didn't look at this properly but it seems like it's somewhat adding something that should really be in Form Wizard 🤔

Reply:
yeah i agree, it does feel that way. i just recall that there was some weirdness i hit during my testing.

Reply:
i remembered one thing! values aren't removed once set (because we don't have invalidation rules) meaning if you choose a radio button and add a conditional comment, then go back and choose a different radio button – the comment remains in `values`. that’s why i read everything from the session model (as in HMPO examples) and then manually delete unnecessary conditionals